### ISSUE TITLE
Allow to discard code when generating code.

### ISSUE THREAD

**Issue by mubbasher** [Microsoft/TypeScript#9137]
I have JavaScript library , where I have multiple inheritance done with my own functions to extend.

But for intellisense purpose we have add some interface to allow lookup for function withing the classes and implemented dummy functions for completeness. But when generating the code, I would like to take that out.

Can we added that feature take out some code when generating JS code.
Just one switch let us say, #NOT_A_CODE #END_NOT_A_CODE ... and ignore the code in between.

**Comment by RyanCavanaugh** [comment]
Interfaces don't generate code; if you need to have a class that doesn't generate code, you can use the `declare` keyword to create an ambient class. There shouldn't be anything you need to do for intellisense / type information purposes that requires a construct that generates code.

**Comment by mubbasher** [comment]
It's not ambient class.
it is actual class.  I get this error.
[ts] 'declare' modifier cannot appear on a class element.

Some stuff needs to be added to the class for intellisense. But It would be only added at runtime.
by some other JS library say for instance.

**Comment by mhegazy** [comment]
not the class member, the whole class.
```ts
declare class C {
    property: number;
    method(a: string): C;    
}


var x = new C();
x.method("string");
```

**Comment by mubbasher** [comment]
Got it that would work in fact.

But if someone ends up in a situation like me...

I just created a small quick code in C# to post process my final javascript and remove the code between the comments, //##NOT_A_CODE //##END_NOT_A_CODE.
My situation is slightly different.

And if someone interested in how to do multiple inheritance using JavaScript or even using the TypeScript which apparently doesn't support it ... give me a ping. :) as I did achieved :+1:
